frontend/javascripts/oxalis/shaders/texture_access.glsl.ts

```diff
@@ -41,50 +41,39 @@ export const getRgbaAtIndex: ShaderModule = {
 };
 export const getRgbaAtXYIndex: ShaderModule = {
   code: `
-    vec4 getRgbaAtXYIndex(sampler2D dtexture, float textureWidth, float x, float y) {
-      return texture2D(
-          dtexture,
-          vec2(
-            (floor(x) + 0.5) / textureWidth,
-            (floor(y) + 0.5) / textureWidth
-          )
-        ).rgba;
-    }
-
     // Define this function for each segmentation and color layer, since iOS cannot handle
     // sampler2D textures[dataTextureCountPerLayer]
     // as a function parameter properly
 
     <% _.each(layerNamesWithSegmentation, (name) => { %>
-      vec4 getRgbaAtXYIndex_<%= name %>(float textureIdx, float textureWidth, float x, float y) {
-        vec2 accessPoint = (floor(vec2(x, y)) + 0.5) / textureWidth;
-
+      vec4 getRgbaAtXYIndex_<%= name %>(float textureIdx, float x, float y) {
         // Since WebGL 1 doesnt allow dynamic texture indexing, we use an exhaustive if-else-construct
         // here which checks for each case individually. The else-if-branches are constructed via
         // lodash templates.
 
         <% if (dataTextureCountPerLayer === 1) { %>
             // Don't use if-else when there is only one data texture anyway
-            return texture2D(<%= name + "_textures" %>[0], accessPoint).rgba;
+
+            return texelFetch(<%= name + "_textures" %>[0], ivec2(x, y), 0).rgba;
```

Comment:
Nice, `texelFetch` is exactly what we need here - glad to see the workaround for this being removed.